### PR TITLE
Week 1 assessment yan

### PR DESCRIPTION
**Pull request by yanxu2021**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -1,54 +1,138 @@
...
 var temp1 = 35
+console.log(temperature(temp1))//Output: 35 is below boiling point.
 var temp2 = 350
+console.log(temperature(temp2))//Output: 350 is above boiling point.
 var temp3 = 212
+console.log(temperature(temp3))//Output: 212 is at boiling point.
+var temp4 = true
+console.log(temperature(temp4))//Output: true is below boiling point....true == 1 
+var temp5 = "yan"
+console.log(temperature(temp5))//Output: Invaild temperature.Cool!
```
Comment (WalkerTexas-coder) [inline code-challenges.js:39]: Epic! Great testing!

File: code-challenges.js [hunk 2]
```diff
@@ -1,54 +1,138 @@
...
 var myNumbers1 = [3, 7, 0, 36, -9]
 var myNumbers2 = [8, -7, 22, 9, 13]
 
+const singleSortArray = (numbers1, numbers2) =>{
+    //return numbers1.concat(numbers2)---->[3,7,0,36,-9,8,-7,22,9,13]
+    //return numbers1.concat(numbers2).sort()---->>[-7,-9,0,13,22,3,36,7,8,9]
+    //It seems .sort() handles the array itms as strings.
+
+    //Googled and Get the method below from stackoverflow:
+    var newNumbers = new Float64Array(numbers1.concat(numbers2))
```
Comment (WalkerTexas-coder) [inline code-challenges.js:59]: Haha awesome!

File: code-challenges.js [hunk 3]
```diff
@@ -1,54 +1,138 @@
...
+}
 var numberExample1 = 42
 var numberExample2 = 27
-
+console.log(subtraction(numberExample1 , numberExample2))
+//42 is greater than 27, the difference is 15.
 var numberExample3 = 7
 var numberExample4 = 19
+console.log(subtraction(numberExample3 , numberExample4))
+//7 is smaller than 19, the difference is 12.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:133]: EXCELLENT!

File: code-challenges.js [hunk 4]
```diff
@@ -1,54 +1,138 @@
...
 var numberExampleRefactor2 = "hello"
+console.log(subtraction(numberExampleRefactor1 ,numberExampleRefactor2 ))
+//Your input is not a number!
+//Forgot to put number with ""! the typeof gonna return a string to compare. 
+//Without "", the output will be undefined.
 
 var numberExampleRefactor3 = 27
 var numberExampleRefactor4 = 24
+console.log(subtraction(numberExampleRefactor3 , numberExampleRefactor4))
+//27 is greater than 24, the difference is 3.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:155]: Great Work!

File: coding-concepts.js [hunk 5]
```diff
@@ -8,44 +8,54 @@
...
-
-// a) Your answer:
-// b) Verify and explain:
+console.log(typeof dataTypes.length)://typeof 3--->output: number
+console.log(dataTypes.length)//Output:3
+console.log(typeof true)// "Boolean"<----output will be a string.
+
+// a) Your answer: Arrey has Length Property. dataTypes.length will be 3. and typeof 3 will be number.  I am not sure typeof here but try. 
+// b) Verify and explain:the output is number. Review: The typeof operator returns the data type of its operand in the form of a string.
+//The operand can be any object, function, or variable. <-----Interesting!!!
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:60]: It's cool right!

File: interview-questions.md [hunk 6]
```diff
@@ -8,59 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  - You don't have to know how to make it to use the function.
+  - The function like a machine supposed to serve every time you want to use it/if the machine doesn't work properly you need figure out by yourself or other people, machine itself can't not figure out what the problem.
 
-  Researched answer:
 
+  Researched answer:
+  **Functions** are a fundamental building block in JavaScript and in programming in general. Functions are reusable pieces of code that only execute when invoked and always return an output. Functions can have information (arguments) passed into the scope of the function through placeholders called parameters. Functions are designed to be used many times in a program and should be as generic as possible while still being descriptive as to their purpose.
+- Functions are little machines that run when we decide and how often we decide.
+- Functions should be relatively neutral.
+- Encapsulation, one of the four pillars of OOP. The function logic is entirely contained in the function and doesn't rely on any external information other than the given input.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:20]: Both answers here are great!

File: interview-questions.md [hunk 7]
```diff
@@ -8,59 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. In regards to functions, what is the difference between an argument and a parameter?
 
-  Your answer:
+  Your answer:Most of times they seem same value but different name, actually they are very different. Argument for function is more like a placeholder you gonna use that to perform the actual operations within the function. Parameter is the specific value to transfer to argument when you need invoke the function. -------->>>>>>My answer is wrong!!! Opposite!!! ARGUMENT IS THE ACTUAL CONTENT! PARAMETER IS LIKE A PLACEHOLDER.  In Chinese, it is easy to tell the difference. 形(means form)参 and 实(means actual)参
+
```
Comment (WalkerTexas-coder) [inline interview-questions.md:26]: Oh that is cool! and yeah it's easy to flip these ideas in english! a lot of people do early on.

File: interview-questions.md [hunk 8]
```diff
@@ -8,59 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+  Researched answer:Every value in an array has a particular location know as the **index**. Indexes are sequential numbers that are a bit like an address for every element in the array. Arrays are zero indexed which means the elements are numbered starting with the number 0 and increase by one whole number for each new element. Index of a string is a particular character.**Value** = element's content. 
 
 
 4. What is a built-in method?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Some operations we already know how to do and we need do that all the time but don't want to repeat(created or defined by yourself everytimes), we can simplified the process to create some built-in method/ function to perform this specific operations so easy just by calling the name.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:44]: Yes!

File: code-challenges.js [hunk 9]
```diff
@@ -1,54 +1,138 @@
...
+    var newString=""//It is very important to put newString outside the loop because We need print out the final string.
+    for(var i=string.length-1; i >= 0; i--){
+        //console.log(string[i])---->Output will be in separate line. 
+        //Return will stop the loop and only return the last letter of the string.
+        
+        newString += string[i]     
+        
+    }
+    return newString//Use return NOT console.log() which will output: undefined!!!
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.js:93]: Amazing!

File: interview-questions.md [hunk 10]
```diff
@@ -8,59 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+  Researched answer:Two peopel collaborating on one computer to collectively produce one program. It is a very collaborative way of working and involves a lot of communication. While a pair of developers work on a task together, they do not only write code, they also plan and discuss their work. They clarify ideas on the way, discuss approaches and come to better solutions.
+  **The Roles** One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in. The two programmers switch roles frequently. Both must understand why and how."How do we talk" is very critical for the whole pair programming process.
 
 
 6. What is something we did in class this week you found helpful?  
 
-  Your answer:
-
+  Your answer: Since I have got some college computer science courses, I do feel LEARN's class is more practical, we have enough time to pair programming , which is collaborative and easy for everyone get involved instead of the boring lecture. Lot appreciates for others' idea\ hardwork.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:59]: We appreciate having your ideas and your hard work!

**Review by WalkerTexas-coder — APPROVED** [review]
Yan, This was so fun to go through! Your Pseudo code is informative and it pumped me up! Let's Go!!! Your logic is great! Your code is readable! and where you were wrong you explained it perfectly! Thank you for teaching me a little Chinese (Mandarin?). I am looking forward to your next assessment. Your PR is approved!

**Review by WalkerTexas-coder — APPROVED** [review]
disregard this text box

**Comment by yanxu2021** [comment]
> Yan, This was so fun to go through! Your Pseudo code is informative and it pumped me up! Let's Go!!! Your logic is great! Your code is readable! and where you were wrong you explained it perfectly! Thank you for teaching me a little Chinese (Mandarin?). I am looking forward to your next assessment. Your PR is approved!

Thank you so much for your hard work!!!

**Comment by yanxu2021** [comment]
> Yan, This was so fun to go through! Your Pseudo code is informative and it pumped me up! Let's Go!!! Your logic is great! Your code is readable! and where you were wrong you explained it perfectly! Thank you for teaching me a little Chinese (Mandarin?). I am looking forward to your next assessment. Your PR is approved!

Yes.. It is Mandarin, please skip it, the only reason for me, note in Chinese === highlighter. LOL Appreciate a lot for your help again~

**Pull request merged**